src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNode.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System.Collections.Generic;
+using System.Collections.Specialized;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class TreeViewNode : DependencyObject
+	{
+		private TreeViewNode _parent;
+		private bool _hasUnrealizedChildren;
+		private object m_itemsSource = null;
+		private ItemsSourceView m_itemsDataSource = null;
+
+		public TreeViewNode()
+		{
+			var collection = new TreeViewNodeVector();
+			collection.SetParent(this);
+			Children = collection;
+			collection.VectorChanged += ChildVectorChanged;
+		}
+
+		public TreeViewNode Parent
+		{
+			get => _parent;
+			internal set
+			{
+				_parent = value;
+				UpdateDepth(value != null ? value.Depth + 1 : -1);
+			}
+		}
+
+		public bool HasUnrealizedChildren
+		{
+			get => _hasUnrealizedChildren;
+			set
+			{
+				_hasUnrealizedChildren = value;
+				UpdateHasChildren();
+			}
+		}
+
+		internal bool IsContentMode { get; set; }
+
+		internal TreeNodeSelectionState SelectionState { get; set; }
+
+		public IList<TreeViewNode> Children { get; }
+
+		private void UpdateDepth(int depth)
+		{
+			// Update our depth
+			SetValue(DepthProperty, depth);
+
+			// Update children's depth
+			foreach (var child in Children)
+			{
+				child.UpdateDepth(depth + 1);
+			}
+		}
+
+		private void UpdateHasChildren()
+		{
+			bool hasChildren = Children.Count != 0 || HasUnrealizedChildren;
+			SetValue(HasChildrenProperty, hasChildren);
+		}
+
+		private void ChildVectorChanged(IObservableVector<TreeViewNode> sender, IVectorChangedEventArgs args)
+		{
+			var collectionChange = args.CollectionChange;
+			var index = args.Index;
+			UpdateHasChildren();
+			RaiseChildrenChanged(collectionChange, index);
+		}
+
+		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'property'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)